docs/testing-with-phpunit.md

```diff
@@ -0,0 +1,225 @@
+# Testing With PHPUnit
+
+[PHPUnit]() is a powerful framework for testing PHP code and applications. The tests that PHPUnit excels at running can be broken down into two fundamental types:
+
+- Unit tests
+- Integration tests
+
+Unit tests can be run against a unit of code in isolation, such as a function or object and require no persistent state.
+
+Integration tests allow you to test a running application. In the case of Altis this type of testing is often more useful however it is possible to run both types of test.
+
+Currently only PHPUnit <= 7.1 is supported.
+
+
+## Setting Up PHPUnit
+
+To set up PHPUnit from scratch run the following command:
+
+```sh
+composer dev-tools scaffold phpunit
+```
+
+This will create the following files:
+
+- `phpunit.xml.dist` - Instructions for PHPUnit.
+- `.tests/bootstrap.php` - Handles loading and installing Altis before running tests.
+- `.tests/setup.php` - Provides access to test suite functions such as `tests_add_filter()`.
+- `.tests/config.php` - A custom config file used to define constants or other code for the test run.
+- `.tests/inc/class-test-sample.php` - A basic test to get you started.
+
+If you already have PHPUnit set up for a different environment you can temporarily move your existing `phpunit.xml.dist` file and a `.tests` directory to compare the scaffolded versions to your existing ones. The dot prefix ensures this directory is not accessible to HTTP requests.
+
+The scaffolded files can be edited as you need.
+
+
+## Running Tests
+
+With the above scaffolding in place running your unit tests is straightforward:
+
+```sh
+composer dev-tools phpunit
+```
+
+By default this will attempt to run your tests on the [Local Server](docs://local-server) environment. If you are using [Local Chassis](docs://local-chassis) add the `--chassis` flag eg:
+
+```sh
+composer dev-tools phpunit --chassis
+```
+
+### Passing Arguments To PHPUnit
+
+Often you may have multiple test directories, different configurations to run or you want to generate coverage reports. To pass any of the supported command line options to PHPUnit you need to add them after the options delimiter `--`. For example:
+
+```sh
+composer dev-tools phpunit -- --coverage-xml coverage --log-junit junit.xml
+```
+
+The [full list of PHPUnit command line options is available here](https://phpunit.readthedocs.io/en/7.1/textui.html) or you can run `composer dev-tools phpunit -- --help`.
+
+
+## Writing Tests
+
+On top of the standard PHPUnit framework Altis bundles the WP PHPUnit framework to provide advanced functionality for integration testing.
```

Comment:
Is the WP PHPUnit framework the Core one? Might be better to link to it to be clear.

Reply:
Ah I meant to do that. It is based on the core one but it's a composer friendly extraction of it.